### DISCUSSION TITLE
how to plot the arrows based on the ebsdid?

### DISCUSSION THREAD

**Discussion by 419278981** [mtex-toolbox/mtex#587] [Ask Anything] [answered]
Based on the GND density, I calculated the energy difference between two EBSD points. Furthermore, I want to draw an arrow between these two points in the IPF diagram. Among them, the length of the arrow can indicate the amount of energy, and the direction of the arrow indicates the direction between pixels.

**Comment by kilir** [comment, marked as answer]
Hi,
I assume you mean you want to draw an arrow between the orientations of these EBSD points. You can scale your arrowhead length by your energy value. I think changing the length of the arrow does not work since in that case it would not connect the orientations at both points
```
mtexdata titanium

id1 = 1045
id2 = 2333

plotIPDF([ebsd('id',id1).orientations ebsd('id',id2).orientations],[xvector,yvector,zvector],'arrow','length',50)
```
Cheers,
Rüdiger

**Reply by 419278981** [reply]
thanks for your answer！
![arrow](https://user-images.githubusercontent.com/32330722/99953906-e9876c00-2dbc-11eb-8d49-e8292199a60c.png)
There may be something wrong with my expression. What I want to achieve is this effect

**Reply by kilir** [reply]
Indeed, "IPF diagram" is understood by most as something else, but in principle it remains the same:
```
plot(ebsd('t'),ebsd('t').orientations)
hold on
arrow([ebsd('id',id1).x ebsd('id',id1).y], [ebsd('id',id2).x ebsd('id',id2).y])
hold off
```
Cheers,
Rüdiger

**Comment by 419278981** [comment]
Yeah, it works very well， thanks！
